dhis-2/dhis-services/dhis-service-setting/src/main/java/org/hisp/dhis/setting/DefaultSystemSettingManager.java

```diff
@@ -194,12 +192,14 @@ public Serializable getSystemSetting( SettingKey key )
      * cache hits.
      */
     @Override
+    @Transactional( readOnly = true )
     public Serializable getSystemSetting( SettingKey key, Serializable defaultValue )
     {
         SerializableOptional value = settingCache.get( key.getName(),
-            k -> getSystemSettingOptional( k, defaultValue ) ).get();
```

Comment:
FYI: It might be hard to see that the last `get()` here does not properly fall back to the `defaultValue`